### PR TITLE
#170230517 Display all notification for a specific user

### PR DESCRIPTION
**Pull request by NiyoEric**
#### What does this PR do?
The PR displays all notifications for a specific user

#####  **Acceptance Criteria**
- Notification for new comment
- Notification for a trip
- Notification for a booking
- Notification for accommodation.

#### How should this be manually tested?
- Open your terminal
- `git clone https://github.com/andela/team-odd-bn-backend.git`
- `cd team-odd-bn-backend`
- `git checkout ft-all-user-notification-170230517`
- Make sure your `.env` file is updated.
- Run migrations `npm run db-migrate`
- Run `psql`  ([Download](https://www.postgresql.org/download/) if it's not installed)
- Run `\c barefoot_nomad_test_db` to connect to the database
- Run `SELECT * FROM {tableName}`
- OR `select * from INFORMATION_SCHEMA.TABLE_CONSTRAINTS;` to see the schema

#### Any background context you want to provide?
No

#### What are the relevant pivotal tracker stories?
[#170230517](https://www.pivotaltracker.com/story/show/170230517)

#### Screenshots (if appropriate)
All notification
Screen Shot 2019-12-11 at 08 50 31

Documentation
Screen Shot 2019-12-11 at 16 14 30

#### Questions:

### CODE REVIEW

File: src/routes/api/userRoute.js [hunk 1]
```diff
@@ -129,4 +129,5 @@ const { isImage } = customValidator;
 */
 userRoute.put('/profile-settings', verifyToken, profileUpdateRules(), checkInputDataError, isImage, isManager, isUserVerified, UserController.updateProfile);
 userRoute.get('/view-profile', verifyToken, isUserVerified, UserController.viewProfile);
+userRoute.get('/notification', verifyToken, isUserVerified, UserController.viewNotification);
```
Comment (hezronkimutai) [inline src/routes/api/userRoute.js:153]: @NiyoEric  please add swagger doc for this route.

**Review by hezronkimutai — COMMENTED** [review]
Nice work @NiyoEric I like the way you have re-used many functionalities and also refactored the existing notification code and this making my work easier, I have left one feedback, please have a look at it.

File: src/routes/api/userRoute.js (hunk 1 above)
Comment (NiyoEric) [reply src/routes/api/userRoute.js:153]: I have added it.

**Review by hezronkimutai — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]
Good job @NiyoEric 
I like your modular coding techniques. I  need to learn from you :)

File: src/services/NotificationService.js [hunk 2]
```diff
@@ -28,26 +29,146 @@ class NotificationService {
...
+    const { managerId } = userProfileInfo.dataValues;
+    const message = `The Trip request no ${params.tripRequestId} has been commented on`;
+    let receiverId;
+
+    if (userId === user.id) {
+      receiverId = managerId; // is a manage
+    } else {
+      receiverId = userId; // is a user
+    }
+
```
Comment (william000000) [inline src/services/NotificationService.js]: @NiyoEric How do you know who you are going to send the notification?

File: src/tests/090-notification.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    chai.request(app)
+      .get('/api/v1/users/notification')
+      .set('token', managerToken)
+      .end((err, res) => {
+        expect(res.status).eql(200);
+        expect(res.body.message).eql('Available notification');
+        expect(res.body).have.property('data');
+        expect(res.body.data).to.be.an('array');
+        done(err);
+      });
```
Comment (william000000) [inline src/tests/090-notification.js:30]: Good Job @NiyoEric, I like the way you have tested your work!

File: src/routes/api/userRoute.js [hunk 4]
```diff
@@ -127,6 +127,28 @@ const { isImage } = customValidator;
 *       500:
 *         description: Internal server error
 */
+
+/**
+ * @swagger
+ *
+ * /users/notification:
+ *    get:
+ *      summary: Available users
```
Comment (william000000) [inline src/routes/api/userRoute.js]: Kindly review the message, you wrote as a summary. I think it might be `Get all available notification`

**Review by william000000 — COMMENTED** [review]
Good Job @NiyoEric, Kindly attend to the feedback I left.

File: src/services/NotificationService.js (hunk 2 above)
Comment (NiyoEric) [reply src/services/NotificationService.js]: If user id saved in trips table equal to user id from token, that means a user adds a comment. The notification should be sent to the manager. Vice versa.

Comment (william000000) [reply src/services/NotificationService.js]: Okay, It makes sense.

**Review by william000000 — APPROVED** [review]

**Review by julietezekwe — APPROVED** [review]

**Pull request merged**